static/js/pivotal.js

```diff
@@ -42,172 +40,212 @@
+            url: data.url,
+            status: data.current_state,
+            dependencies: list
+          });
+        });
       }
-
-      matches.forEach(function(ticket, i, arr) {
-        var id = stripBrackets(ticket);
-        if (id && !(id in pivotal_ids)) {
-          pivotal_ids[id] = true;
-        }
-      });
-    }
-
-    return Object.keys(pivotal_ids); // returning just the keys gives us all unique pivotal IDs
+    });
   }
 
-  function getPivotalInfo(pt_token, story_id, callback){
+  /**
+   * getRepoDependencies return a list of dependencies for stroy in a project
+   * @param  {String}   pt_token   Pivotal token
+   * @param  {number}   project_id Project ID
+   * @param  {number}   story_id   Story ID
+   * @param  {Function} callback   Returns Pivotal story ID list
+   */
+  function getRepoDependencies(pt_token, project_id, story_id, callback) {
     $.ajax({
-      url: config.urls.pivotal.base + config.urls.pivotal.api + '/stories/' + story_id,
+      url: config.urls.pivotal.api + '/projects/' + project_id + '/stories/' + story_id + '/comments',
       type: 'GET',
-      beforeSend: function(request) {
-        request.setRequestHeader('X-TrackerToken', pt_token);
+      headers: {
+        'X-TrackerToken': pt_token
       },
       success: function(data) {
-        var info = {
-          'id': story_id
-        };
-        info['status'] = data['current_state'];
-        info['url'] = data['url'];
-        callback(info);
+        // Find all git commit messages
+        var list = data.filter(function(obj) {
+          return obj.commit_type;
+        });
+        // Find repo name from comment body text
+        list = list.map(function(item) {
+          return item.text.split('https://github.com/gengo/')[1].split('/commit/')[0];
+        });
+
+        callback(removeDupesFromArray(list));
       }
     });
   }
 
-  var diffs = getProjectDiffs();
-  setTimeout(function() {
-    diffs = getProjectDiffs();
-  }, 1000);  // fetch diffs again after 1 sec
+  /**
+   * onGetPivotalStoryInfoComplete is triggered when pitotal info loading completted,
+   * and building up HTML to insert into the Story block.
+   * @param  {String} project   GitHub project name
+   * @param  {Array} storyList  Pivotal story info array
+   */
+  function onGetPivotalStoryInfoComplete(project, storyList) {
+    var html = storyList.map(function(story) {
+      return '<div> \
+                <a href="'+ story.url +'" target="_blank">#'+ story.id +'</a> \
+                &nbsp; \
+                <span class="label label-'+ mapStatusLabelClass(story.status) +'">'+ story.status +'</span> \
+                &nbsp; \
+                <span class="badge" data-toggle="popover" data-content="<li>'+ story.dependencies.join('</li><li>') +'</li>">'+ story.dependencies.length +'</span> \
+             </div>';
+    });
+
+    $('.project[data-id="'+ project +'"]')
+      .find('.story')
+        .append(html.join(''))
```

Comment:
Yeah, it works! Switched from `.append()` to `.html()` which think make more sense here, checked that it work too :santa: